src/EditorFeatures/Test2/Rename/InlineRenameTests.vb

```diff
@@ -2130,5 +2130,71 @@ class [|C|]
                 Await VerifyTagsAreCorrect(workspace, "BarGoo")
             End Using
         End Function
+
+        <WpfTheory>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function RenameTupleElement_FromDeclaration(host As RenameTestHost) As Task
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true" LanguageVersion="9.0">
+                            <Document>
+                                class C
+                                {
+                                    void M()
+                                    {
+                                        var g = ([|$$Goo|]: 1, Bar: 2);
+                                        var gb = g.[|Goo|];
+                                    }
+                                }
+                            </Document>
+                        </Project>
+                    </Workspace>, host)
+
+                Dim session = StartSession(workspace)
+
+                ' Type a bit in the file
+                Dim caretPosition = workspace.Documents.Single(Function(d) d.CursorPosition.HasValue).CursorPosition.Value
+                Dim textBuffer = workspace.Documents.Single().GetTextBuffer()
+
+                textBuffer.Insert(caretPosition, "Bar")
+
+                session.Commit()
+
+                Await VerifyTagsAreCorrect(workspace, "BarGoo")
+            End Using
+        End Function
+
+        <WpfTheory>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function RenameTupleElement_FromUse(host As RenameTestHost) As Task
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true" LanguageVersion="9.0">
+                            <Document>
+                                class C
+                                {
+                                    void M()
+                                    {
+                                        var g = ([|Goo|]: 1, Bar: 2);
+                                        var gb = g.[|$$Goo|];
```

Comment:
Do we have a test if the name of this member is inferred?

Reply:
That's a very good point, thank you. We already have a scenario where we break things if people are using inferred names, but we do correctly mark conflicts which this change does not, so I will see if I can do that easily without fully implementing the hard bits.